libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/HttpRequest.cs

```diff
@@ -282,62 +282,40 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
 
                 switch (Method)
                 {
                     case HttpMethod.POST:
-                        if (instanceBody == null)
-                        {
-                            response = await client.SendAsync(request, cancellationToken).ConfigureAwait(false);
-                        }
-                        else
-                        {
-                            using var postContent = new StringContent(instanceBody.ToString(), Encoding.UTF8, contentType);
-                            traceInfo.request.content = instanceBody.ToString();
-                            request.Content = postContent;
-                            traceInfo.request.headers = JObject.FromObject(request.Content.Headers.ToDictionary(t => t.Key, t => (object)t.Value?.FirstOrDefault()));
-                            response = await client.SendAsync(request, cancellationToken).ConfigureAwait(false);
-                        }
-
-                        break;
-
                     case HttpMethod.PATCH:
-                        if (instanceBody == null)
-                        {
-                            response = await client.SendAsync(request, cancellationToken).ConfigureAwait(false);
-                        }
-                        else
-                        {
-                            using var patchContent = new StringContent(instanceBody.ToString(), Encoding.UTF8, contentType);
-                            request.Content = patchContent;
-                            traceInfo.request.content = instanceBody.ToString();
-                            traceInfo.request.headers = JObject.FromObject(request.Content.Headers.ToDictionary(t => t.Key, t => (object)t.Value?.FirstOrDefault()));
-                            response = await client.SendAsync(request, cancellationToken).ConfigureAwait(false);
-                        }
-
-                        break;
-
                     case HttpMethod.PUT:
                         if (instanceBody == null)
                         {
                             response = await client.SendAsync(request, cancellationToken).ConfigureAwait(false);
                         }
                         else
                         {
-                            using var putContent = new StringContent(instanceBody.ToString(), Encoding.UTF8, contentType);
-                            traceInfo.request.content = instanceBody.ToString();
-                            request.Content = putContent;
-                            traceInfo.request.headers = JObject.FromObject(request.Content.Headers.ToDictionary(t => t.Key, t => (object)t.Value?.FirstOrDefault()));
-                            response = await client.SendAsync(request, cancellationToken).ConfigureAwait(false);
+                            if (instanceBody.GetType() == typeof(byte[]))
+                            {
+                                using var postContent = new ByteArrayContent((byte[])instanceBody);
```

Comment:
since now, we are using put, patch and post methods together, we can rename the `postContent` var to something like `content`?

Reply:
That's true, nice catch. Thanks!